Makefile

```
@@ -0,0 +1,41 @@
+REPO_NAME = operating-systems
+IMAGE_NAME = $(REPO_NAME)/docusaurus:latest
+CONTAINER_NAME = open-edu-hub-$(REPO_NAME)-bash
+OUTPUT_DIR = $$PWD/.output/$(REPO_NAME)
+
+.PHONY: all buildimg build serve run_bash enter_bash stop_bash clean cleanall
+
+all: build
+
+buildimg:
+	docker build -f ./Dockerfile --tag $(IMAGE_NAME) .
+
+build: buildimg
+	@echo "Building content. This will take a while (several minutes) ..."
+	@echo "After the build, run"
+	@echo ""
+	@echo "       make serve"
+	@echo ""
+	@mkdir -p $(OUTPUT_DIR)
+	docker run --rm -v $$PWD/:/content -v $(OUTPUT_DIR):/output $(IMAGE_NAME)
+
+serve:
+	@echo "Point your browser to http://localhost:8080/$(REPO_NAME)"
+	@cd $(OUTPUT_DIR)/.. && python3 -m http.server 8080
+
+run_bash: buildimg
+	@mkdir -p $(OUTPUT_DIR)
+	docker run -d -it --entrypoint /bin/bash --name $(CONTAINER_NAME) -v $$PWD/:/content -v $(OUTPUT_DIR):/output $(IMAGE_NAME)
```

```suggestion
	docker run -d -it --rm --entrypoint /bin/bash --name $(CONTAINER_NAME) -v $$PWD/:/content -v $(OUTPUT_DIR):/output $(IMAGE_NAME)
```
Is there any use for keeping the container around after it is done?

---

There may be in case you want to debug it or investigate something else. Let's be conservative and not remove the container.